src/content/language/collections.md

```diff
@@ -242,78 +250,180 @@ final constantMap = const {2: 'helium', 10: 'neon', 18: 'argon'};
+```
+
+To learn more about the operator used in spread operations
+for collections, see [Spread operator][].
+
+[Spread operator]: /language/operators/#spread-operators
+
+### Null-aware spread operations
+
+With a null-aware spread operation, you can add values from
+one collection to another collection. Null values inside of
+a collection can be passed in, but null collections are
+ignored. A null-aware spread operation has this
+signature in a collection:
+
+```dart
+...?<collection_name>
+```
+
+In the following example, `a` is ignored because it's
+null, but `b` is passed into `items`.
+
+```dart
+var a = null;
+var b = [1, 2, null];
+var items = [0, ...?a, ...?b, 4]; // [0, 1, 2, null, 4]
+```
+
+To learn more about the operator used in null-aware spread
+operations for collections, see [Spread operator][].
+
+[Spread operator]: /language/operators/#spread-operators
 
 <a id="collection-operators"></a>
-### Control-flow operators
 
-Dart offers **collection if** and **collection for** for use in list, map,
-and set literals. You can use these operators to build collections using
-conditionals (`if`) and repetition (`for`).
+### If statements
+
+You can use an `if` statement inside of a collection to
+conditionally include values in the collection based on a
+boolean expression. An `if` statement has this signature in
+a collection:
 
-Here's an example of using **collection if**
-to create a list with three or four items in it:
+```dart
+// boolean expression must be true to include result
+if (<bool_expression>) <result>
+```
 
-<?code-excerpt "misc/test/language_tour/built_in_types_test.dart (list-if)"?>
 ```dart
-var nav = ['Home', 'Furniture', 'Plants', if (promoActive) 'Outlet'];
+// boolean expression must be false to include result
+if (!<bool_expression>) <result>
 ```
 
-Dart also supports [if-case][] inside collection literals:
+In the following example, `1` is included in `items`
+because `includeItem` evaluates as true, but `2` is not
+included because `!includeItem` evaluates as false.
 
 ```dart
-var nav = ['Home', 'Furniture', 'Plants', if (login case 'Manager') 'Inventory'];
+var includeItem = true;
+var items = [0, if (includeItem) 1, if (!includeItem) 2, 3]; // [0, 1, 3]
 ```
 
-Here's an example of using **collection for**
-to manipulate the items of a list before
-adding them to another list:
+To learn more about the `if` statement, see
+[`if` statement][].
+
+[`if` statement]: /language/branches#if
+
+### If case statements
+
+You can use an `if-case` statement inside of a collection to
+conditionally include values in the collection if the
+matching expressions evaluate as true. An `if-case`
+statement has this signature in a collection:
 
-<?code-excerpt "misc/test/language_tour/built_in_types_test.dart (list-for)"?>
 ```dart
-var listOfInts = [1, 2, 3];
-var listOfStrings = ['#0', for (var i in listOfInts) '#$i'];
-assert(listOfStrings[1] == '#1');
+if (<expression> case expression_literal) <result>
```

Comment:
The thing after `case` here is a [pattern](https://dart.dev/language/patterns), not an expression. (It's a little confusing because there are patterns that are simply constant expressions, so you can use an expression here. But you can also use things that aren't expressions. And some things that syntactically look like expressions aren't because pattern syntax overlaps expression syntax.)

Reply:
This is pattern? Nice! I had no idea. I'll update and add a note about that.